plutus-ledger-constraints/src/Ledger/Constraints/OffChain.hs

```diff
@@ -533,32 +534,27 @@ addOwnInput ScriptInputConstraint{icRedeemer, icTxOutRef} = do
     unbalancedTx . tx . Tx.inputs %= (Typed.tyTxInTxIn txIn :)
     valueSpentInputs <>= provided vl
 
--- | Add a typed output and return its value.
+-- | Convert a @ScriptOutputConstraint@ into a @TxConstraint@.
 addOwnOutput
     :: ( MonadReader (ScriptLookups a) m
-        , MonadState ConstraintProcessingState m
-        , FromData (DatumType a)
-        , ToData (DatumType a)
         , MonadError MkTxError m
+        , ToData (DatumType a)
         )
     => ScriptOutputConstraint (DatumType a)
-    -> m ()
-addOwnOutput ScriptOutputConstraint{ocDatum, ocValue} = do
+    -> m TxConstraint
+addOwnOutput ScriptOutputConstraint{ocDatum, ocValue, ocInlineScriptHash} = do
     ScriptLookups{slTypedValidator} <- ask
     inst <- maybe (throwError TypedValidatorMissing) pure slTypedValidator
-    let txOut = Typed.makeTypedScriptTxOut inst ocDatum ocValue
-        dsV   = Datum (toBuiltinData ocDatum)
-    unbalancedTx . tx . Tx.outputs %= (Typed.tyTxOutTxOut txOut :)
-    unbalancedTx . tx . Tx.datumWitnesses . at (P.datumHash dsV) .= Just dsV
-    valueSpentOutputs <>= provided ocValue
+    let dsV = Datum (toBuiltinData ocDatum)
+    pure $ MustPayToOtherScript (Typed.tvValidatorHash inst) Nothing dsV ocInlineScriptHash ocValue
```

Comment:
So were not using anymore functions from TypedValidator like `makeTypedScriptTxOut`? Does that make them obsolete? Or is it because they were not updated to V2.TxOut?

Reply:
Yes, as far as I'm concerned there's not much value to having `TypedScriptTxOut`. Shall I remove it in this PR?